functions/transformations/group.go

```diff
@@ -216,3 +217,27 @@ func (t *groupTransformation) UpdateProcessingTime(id execute.DatasetID, pt exec
 func (t *groupTransformation) Finish(id execute.DatasetID, err error) {
 	t.d.Finish(err)
 }
+
+// `MergeGroupRule` merges two group operations and keeps only the last one
+type MergeGroupRule struct{}
+
+func (r MergeGroupRule) Name() string {
+	return "MergeGroupRule"
+}
+
+// returns the pattern that matches `group |> group`
+func (r MergeGroupRule) Pattern() plan.Pattern {
+	return plan.Pat(GroupKind, plan.Pat(GroupKind, plan.Any()))
+}
+
+func (r MergeGroupRule) Rewrite(node plan.PlanNode) (plan.PlanNode, bool, error) {
+	firstGroup := node.Predecessors()[0]
+	newSpec := node.ProcedureSpec().Copy()
+
```

Comment:
I think it would be safest to not apply this rule unless the group mode is `GroupModeBy` or `GroupModeExcept`.  Right now, those are the only group modes possible, but that could change in the future.  There has been some discussion of allowing users a way to express "group by everything the input groups by, but add or subtract these columns."  So I think adding a check here would guard against wrong answers in the future.

Reply:
Ok, now I got the last lines in the issue description about schema introspection 👍

Reply:
I only checked for the second (last) group not to be by or except, because, in the case of a relative (or fancy) grouping, a subsequent one will always override its behavior. Do you agree?

Reply:
Yes, makes perfect sense.